File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    var tot=0;
    while(1){
        var process=priorities.shift();
        
        if(priorities.some(each => each>process)){
            priorities.push(process);
            location-=1;
            if(location<0) location=priorities.length -1;
            continue;
        }   
        tot++;
        if(location===0) return tot;     
        location-=1;
        if(location<0) location=priorities.length -1;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

// bool compare(int a, int b){
//     return a>b;
// }

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int max=-1;
    queue<pair<int,int>> q;
    priority_queue<int,vector<int>> maxNum;
    for(int i=0;i<priorities.size();i++){
        // if(max<priorities[i]){
        //     max=priorities[i];
        // }
        q.push({priorities[i],i});
        maxNum.push(priorities[i]);
    }
    // sort(maxNum.begin(),maxNum.end(),compare);
    int priority=1;
    max=maxNum.top();
    maxNum.pop();
    while(!q.empty()){
        int cmpValue = q.front().first;
        int cmpIdx = q.front().second;
        q.pop();
        if(cmpValue<max){
            q.push({cmpValue,cmpIdx});
        }else{
            priorities[cmpIdx]=priority;
            priority++;
            max=maxNum.top();
            maxNum.pop();
        }
    }
    return priorities[location];
}